### ISSUE TITLE
Allow switching between deep and shallow references

### ISSUE THREAD

**Issue by Namnodorel** [TouK/krush#42]
This is a small follow-up after having used krush with self-references supported for a while. Although I'm happy with having code generated that can generate objects that reference others of the same kind, this has also started to become a bit of a problem: 

When using the deep/real reference mode of krush, it requires the entire table of self-referential entities to be loaded into memory (because AFAIK there is no way to construct a query in a way that will select a row and also all of its (potentially n layers deep) referenced rows in the same layer). That by itself is a compromise I'm willing to make when I need these references to be correct - but that is not always the case. In my application, there are plenty of cases where only surface-level information is required to do what is needed, and in those cases, loading everything in the table feels a bit wasteful.

That's why I'd like to revisit [this discussion](https://github.com/TouK/krush/pull/11#issuecomment-652844794) - currently, there is only one global option for switching between shallow/deep references (which happens during code generation). 
I would, however, find it useful if there was an (optional) parameter that could be used to turn this on or off per query that is being executed (at runtime). Existing code would automatically use the default value which enables the old, default behavior while new code could make use of being more specific about which mode to use.

Basically, something like this:
````
fun Iterable<ResultRow>.toEntityMap(deepReferences: Boolean = false): MutableMap<Int, Entity> {
  if(deepReferences) {
     // Generate deep references
  } else {
      // Business as usual
  }
}
````

**Comment by Namnodorel** [comment]
This is made obsolete due to the recent PR.
